holidays/countries/taiwan.py

```diff
@@ -91,6 +92,20 @@ def __init__(self, *args, **kwargs):
         kwargs.setdefault("observed_rule", SAT_TO_PREV_WORKDAY + SUN_TO_NEXT_WORKDAY)
         super().__init__(*args, **kwargs)
 
+    def _get_weekend(self, dt: date) -> set[int]:
+        if dt.year >= 2001:
+            return {SAT, SUN}
+        else:
+            # 1998–2000: Sundays as well as the 2nd & 4th Saturdays of each Month.
+            weekend = {SUN}
+            if dt.weekday() == SAT:
+                saturdays = [
+                    week[SAT] for week in monthcalendar(dt.year, dt.month) if week[SAT] != 0
+                ]
+                if dt.day in (saturdays[1], saturdays[3]):
```

Comment:
```suggestion
                second_sat = _get_nth_weekday_of_month(2, SAT, dt.month, dt.year)
                fourth_sat = _get_nth_weekday_of_month(4, SAT, dt.month, dt.year)
                if dt in {second_sat, fourth_sat}:
```
Some optimization. (We can also directly inline `second_sat` and `fourth_sat` or make `fourth_sat = _timedelta(second_sat, 14)`, but this won't give us much).